### PR TITLE
add state to secrets output

### PR DESCRIPTION
**Pull request by GregHanson**
add state to better capture any certs that may be in an unhealthy state

### CODE REVIEW

**Comment by GregHanson** [comment]
feedback from https://github.com/istio/istio/pull/43907#issuecomment-1467256344

**Review by hanxiaop — COMMENTED** [review]
lgtm

File: src/admin.rs [hunk 1]
```diff
@@ -161,10 +162,23 @@ fn dump_cert(x509: &X509) -> CertDump {
...
-        .collect_certs(|id, certs| CertsDump {
-            identity: id.to_string(),
-            ca_cert: [dump_cert(certs.x509())],
-            cert_chain: certs.iter_chain().map(dump_cert).collect(),
+        .collect_certs(|id, certs, state| {
+            let mut dump = CertsDump {
+                identity: id.to_string(),
+                state,
+                ca_cert: [CertDump {
+                    pem: "".to_string(),
```
Comment (ymesika) [inline src/admin.rs:170]: Maybe add a `Default` annotation to the struct instead of setting empty strings?

**Review by ymesika — APPROVED** [review]
LGTM

nit comment

File: src/admin.rs (hunk 1 above)
Comment (GregHanson) [reply src/admin.rs]: done

File: src/admin.rs [hunk 2]
```diff
@@ -161,10 +162,20 @@ fn dump_cert(x509: &X509) -> CertDump {
...
     let mut dump = cert_manager
-        .collect_certs(|id, certs| CertsDump {
-            identity: id.to_string(),
-            ca_cert: [dump_cert(certs.x509())],
-            cert_chain: certs.iter_chain().map(dump_cert).collect(),
+        .collect_certs(|id, certs, state| {
+            let mut dump = CertsDump {
+                identity: id.to_string(),
+                state,
+                ca_cert: [CertDump {
```
Comment (howardjohn) [inline src/admin.rs:169]: nit: do we just want empty vec if there is none?

**Review by howardjohn — APPROVED** [review]

File: src/admin.rs (hunk 2 above)
Comment (GregHanson) [reply src/admin.rs:169]: cc @qfel 

The `ca_cert` field is an array (with set size) and not a Vec, I know Piotr did this to be in line with Envoy API

**Pull request merged**